prow/gcsupload/options.go

```diff
@@ -61,11 +61,18 @@ type Options struct {
 	// paths that are parsed to get more granular
 	// fields.
 	gcsPath gcs.Path
+
+	// LocalOutputDir specifies a directory where files should be copied INSTEAD of uploading to GCS.
+	// This option is useful for testing jobs that use the pod-utilities without actually uploading.
+	LocalOutputDir string `json:"local_output_dir,omitempty"`
```

Comment:
I think I'd prefer us to implement an option on jobs to not do any GCS interaction instead of running a single-purpose tool (`gcsupload`) to do something explicitly not related to GCS or uploads :D

Reply:
The reasoning for adding a local mode to the gcsupload utility instead of having a separate localoutput utility is to minimize skew between local testing and prod behavior. By reusing gcsupload we ensure we output the same files we would upload and perform all the same logic. Minimizing skew between local testing and prod is the motivation for this PR so I'm trying to prioritize that.
I don't really see this as not related to GCS uploads either, I see this more as a dry-run equivalent. It is supposed to simulate a GCS upload.